### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by keith**
Here's some quick stuff I did to get everything working.
- Enforce tabs on the project level
- Clean up some syntax
- Fix a button sender issue
- Fix some compile errors/warnings

This does not include the submodule changes I had to do in Puissant to get everything working.

### CODE REVIEW

File: Mail/DMComposeWindowController.m [hunk 1]
```diff
@@ -110,13 +110,13 @@ - (void)_loadWebview {
...
 - (IBAction)toggleBoldSelectedText:(NSControl *)sender {
-	NSMenuItem *boldMenu = [[NSApp delegate]boldButton];
+	NSMenuItem *boldMenu = [(DMAppDelegate *)[NSApp delegate] boldButton];
 	[boldMenu.menu cancelTracking];
 	[boldMenu.menu performActionForItemAtIndex:1];
 }
 
 - (IBAction)toggleItalicSelectedText:(id)sender {
-	NSMenuItem *boldMenu = [[NSApp delegate]boldButton];
+	NSMenuItem *boldMenu = [(DMAppDelegate *)[NSApp delegate]boldButton];
```
Comment (CodaFi) [inline Mail/DMComposeWindowController.m]: just as you did above, an extra space here

```objective-c
NSMenuItem *boldMenu = [(DMAppDelegate *)[NSApp delegate] boldButton];
```

Comment (Keithbsmiley) [inline Mail/DMComposeWindowController.m]: Yep cool. Missed that the first time.

**Comment by CodaFi** [comment]
There's so much stuff in here that has to change one way or another.  Thanks for taking the first steps.

:boat:

**Comment by Keithbsmiley** [comment]
Definitely. Like I mentioned above, that Puissant submodules update was required before I could build anything.

**Comment by CodaFi** [comment]
Welp, I've managed to temporarily fix the godawful mess that is the submodules.  It should be all good using the stock instructions with a clean clone.

**Pull request merged**